networkx/algorithms/d_separation.py

```diff
@@ -140,3 +160,190 @@ def d_separated(G, x, y, z):
         return False
     else:
         return True
+
+
+@not_implemented_for("undirected")
+def minimal_d_separator(G, u, v):
+    """Compute a minimal d-separating set between X and Y.
+
+    A d-separating set in a DAG is a set that when conditioned on
+    blocks all paths between the two sets of variables. This function
+    constructs a set that is "minimal", meaning it is the smallest
+    d-separating set between the two variables. This is not necessarily
+    unique. For more details, see Notes.
```

Comment:
Can you define the words for readers who don't know this lingo?  Is a d-separating set in a DAG a set of nodes, or edges, or something else? What does the phrase "conditioned on blocks" mean? What is a variable? and what two sets of variables are you talking about?  This function constructs a (add d-separating) set. And what does "between the two variables" mean? 

Also the first line of the doc_string refers to  X and Y. We should be clear in that first line about whether u & v are nodes or sets of nodes. The rest of the docs need to be updated for u and v also.

Below, it talks about a "d-separation test", but that language is not used elsewhere. Also, "is commonly used with" begs the question as to whether it *can* be used with other types of graphs. Is it only defined for DAGs? Could we extend this function to work with cyclic or even undirected graphs?  Maybe the second sentence (or an edited version) would be sufficient to describe the NetworkXError.

Reply:
I have reworded both function. Lmk wdyt?